docs/topics/iotea-components.md

```diff
@@ -8,68 +8,132 @@
 
-- Transforms and validates incoming events, configurable as ETL-Pipeline
-- Multi-Channel supports an arbitary amout of paralell ETL-Pipelines
-- Completly configurable via config files (JSONata transformations) and JSON Schema validations
-- Validates value type against metadata (boolean, string, number, object, any)
-- Validates type against metadata using Metadata-Manager Slave
-- Freely scalable by ingestion instance count
+These components can be deployed on the edge device itself which reduces traffic to cloud (for [Talents](#Talent) running in the cloud) to minimum using [_ProtocolGateway_](#ProtocolGateway)-based routing.
 
-### Encoding
+## ConfigManager
 
-- Encodes incoming events by various encoders (minmax, delta, categorical) --> Not implemented yet
-- Uses Metadata-Manager Slave to retrieve metadata
-- Uses Instance-Manager to store an encoded field into the appropriate instance
-- Freely scalable by ingestion instance count
+The ConfigManager manages and provides the platform configuration. This is done statically via a configuration file (_config.json_).
 
-### Routing
+The ConfigManager provides a periodic [_Talent_](#Talent) discovery and provides platform configuration (e.g. available _Types_ and _Features_) and dynamically via a periodic _Talent_ discovery. It holds the [_MetadataManager_](#MetadataManager) and [TalentConfigManager](#TalentConfigManager) master instances and synchronize all other instances of them in the system.
 
-- Freely scalable by ingestion instance count
-- Uses rules to determine whether an event should be forwareded to a specific talent
-  - Rules can be any type of boolean algebra (containing and/or rules)
-  - Enriches event with all fields a talent is "interested" in
+## Ingestion
 
-### Talent
+This component is used to transform and validate incoming events and it's configurable as an ETL-pipeline and Multi-Channel supports an arbitrary amount of parallel ETL-pipelines and its completely configurable via configuration files (json) and scalable via multiple instances.
 
-- Supports local and cloud talents (using FAAS like Lambda functions)
-  - Automatically syncs to cloud message brokers via ProtocolGateway
+It validates value types (boolean, string, number, object, any) and _Types_ against metadata (e.g. using [_MetadataManager_](#MetadataManager)).
+
+## Encoding
+
+This component encodes incoming events by various encoders (minmax, delta, categorical) and it uses the InstanceManager to store an encoded field into the appropriate instance. It's scalable via multiple instances.
+
+## Routing
+
+This component routes the events to specific [_Talents_](#Talent) by evaluating their related rule-set. It's scalable via multiple instances.
+
+## Sub Components
+
+This components are used by the other platform components and act like helper or has been extracted to make them re-usable.
```

Comment:
This components --> These components

has been extracted --> have been extracted

Reply:
Done